docs/tutorials/generic_field_type/7_add_a_validation.md

```diff
@@ -1,35 +1,9 @@
 # Step 7 - Add basic validation
 
-!!! tip
-
-    You can find all files used and modified in this step on [GitHub](https://github.com/ezsystems/generic-field-type-tutorial/tree/Step_7).
-
 To provide basic validation that ensures both coordinates are provided, add assertions to the `src/FieldType/Point2D/Value.php`:
 
 ```php
-<?php
-
-use Symfony\Component\Validator\Constraints as Assert;
-
-final class Value implements ValueInterface
-
-{  
-   /**
-     * @var float|null
-     *
-     * @Assert\NotBlank()
-     */
-    private $x;
-     
-    /**
-     * @var float|null
-     *
-     * @Assert\NotBlank()
-     */
-    private $y;
-    
- // ...   
-}
+[[= include_file('code_samples/field_types/2dpoint_ft/src/FieldType/Point2D/Value.php', 6, 23) =]]
```

Comment:
```suggestion
[[= include_file('code_samples/field_types/2dpoint_ft/src/FieldType/Point2D/Value.php', 6, 23) =]]
// ...
```
We're not quoting the full file here (therefore, "add" in the introductory text), and the parenthesis goes after the rest of the code. But let's add an ellipsis.